proc/proc.go

```diff
@@ -46,6 +46,7 @@ type Process struct {
 	packageMap map[string]string
 
 	allGCache               []*G
+	dwarfLock               sync.Mutex
```

Comment:
Ideally we create a reader on init of the `Process` struct after we've parsed all the dwarf info we need. After that, any area in `proc` that was creating a new reader should just use the one stored in the `Process` struct. And then, any reader method will have a `lock/defer unlock` step before any work is done. Maybe we also have a method on `Process` like `GetDwarfReader` or something as a convenience method that will `Seek` the reader so caller doesn't have to worry about it?